### ISSUE TITLE
Crashing in some condition

### ISSUE THREAD

**Issue by tim77** [phase1geo/Minder#154]
**minder: node_set_node_info(): com.github.phase1geo.minder killed by SIGSEGV**

```
Version-Release number of selected component:
minder-1.8.0-1.fc32

Additional info:
reporter:       libreport-2.13.1
backtrace_rating: 4
cgroup:         0::/user.slice/user-1000.slice/user@1000.service/gnome-launched-com.github.phase1geo.minder.desktop-5848.scope
cmdline:        com.github.phase1geo.minder
crash_function: node_set_node_info
executable:     /usr/bin/com.github.phase1geo.minder
journald_cursor: s=b22306b984bd4e71a58cab972a25cd56;i=2cac4;b=2412067036c647edaf9318d21d507c58;m=80c9ed89;t=5a7a4bcaba7c0;x=989195a499cb6360
kernel:         5.6.16-300.fc32.x86_64
rootdir:        /
runlevel:       N 5
type:           CCpp
uid:            1000

Truncated backtrace:
Thread no. 1 (10 frames)
 #0 node_set_node_info at com.github.phase1geo.minder
 #2 undo_node_attach_real_undo at com.github.phase1geo.minder
 #3 undo_buffer_undo at com.github.phase1geo.minder
 #4 main_window_do_undo at com.github.phase1geo.minder
 #5 _g_closure_invoke_va at ../gobject/gclosure.c:873
 #8 gtk_button_clicked at gtkbutton.c:1541
 #9 gtk_button_do_release at gtkbutton.c:1845
 #11 gtk_real_button_released at gtkbutton.c:1963
 #16 multipress_released_cb at gtkbutton.c:666
 #17 _gtk_marshal_VOID__INT_DOUBLE_DOUBLEv at gtkmarshalers.c:4803
```
[Retrace](https://retrace.fedoraproject.org/faf/reports/bthash/262db1fbf1a487072cb84580c482cfd5161cabe1).

Downstream report: https://bugzilla.redhat.com/show_bug.cgi?id=1845485

If this not enough information for debug, please tell me. I'll ask original reported or send you all attachments since bug is private.

**Comment by phase1geo** [comment]
I’ll try something simple, but I’m guessing that I will need to know the sequence of changes to determine the cause of the segfault. All I know from this report is that the user clicked on the undo button which undid a node attachment. I’ll look into this problem this evening.

**Comment by phase1geo** [comment]
Yes, I am unable to reproduce this issue. I'm going to some extra instruction in how to reproduce this problem.

**Comment by tim77** [comment]
Waiting for OP. Probably safe to make bug public, but let's wait for OP a little bit.

**Comment by Lokawn** [comment]
I think I am getting the same error on arch linux with ver 1.10.0 and  1.11.0. 
Ver 1.9.0 is working fine.
I dont know how to submit bugreport, but when I run minder from terminal it gives `Segmentation fault (core dumped)` error.
On using strace it gives 
```
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=NULL} ---
+++ killed by SIGSEGV (core dumped) +++
Segmentation fault (core dumped)
```

**Comment by phase1geo** [comment]
@Lokawn Does version 1.11.1 address this problem?  If not, it would be good to run Minder under gdb so that we can get a stacktrace of the crash.

**Comment by Lokawn** [comment]
Same error in ver 1.11.1
[Backtrace for "./app" build](https://paste.ubuntu.com/p/cCJFXDpP5M/)
[Backtrace for makepkg build](https://paste.ubuntu.com/p/3schfjm2v7/)

**Comment by phase1geo** [comment]
I have checked in some changes to the master branch to help deal with this issue.  Is there any way that you can try grabbing the GitHub source code and running from that to see if the issue still persists?

**Comment by Lokawn** [comment]
still getting same error.
[new_trace.log](https://github.com/phase1geo/Minder/files/5291313/new_trace.log)

**Comment by phase1geo** [comment]
I have updated the master branch once more in another attempt to circumvent a segmentation fault on launch.  Please give this another try.  If this doesn't work either, you will need to delete ~/.local/share/minder/custom_themes.xml and relaunch Minder.  This will effectively remove any custom themes that you have previously created and keep the application from crashing on startup.

I would appreciate it if you could try the latest master branch change prior to removing the themes file and let me know the result.

**Comment by Lokawn** [comment]
Build with new sources - App Didn't work. 
[trace1.log](https://github.com/phase1geo/Minder/files/5298578/trace1.log)
 
Remove ~/.local/share/minder/custom_themes.xml & ~/.local/share/minder/tab_state.xml (files listed here used the custom theme) & then start - App is Working.
[custom_themes.xml](https://github.com/phase1geo/Minder/files/5298599/custom_themes.xml.txt)

Going to try this with ver 1.11.1.

**Comment by phase1geo** [comment]
@Lokawn The error reported above is different from the problem listed above.  Could you try running the latest master branch code by downloading it from GitHub and running './app debug' to run the application under gdb.  I'm interested in the stack trace when the application crashes in your case.

**Comment by Lokawn** [comment]
Restored original custom_theme.xml, then executed `./app debug`
[stacktrace.txt](https://github.com/phase1geo/Minder/files/5304701/stacktrace.txt)
